docs/guide/generators.md

```diff
@@ -0,0 +1,100 @@
+---
+layout: default
+title: Generators
+parent: Building ViewComponents
+---
+
+# Generators
+
+The generator accepts a component name and a list of arguments.
+
+This creates an `ExampleComponent` with `title` and `content` attributes:
+
+```console
+bin/rails generate component Example title content
+
+      create  app/components/example_component.rb
+      invoke  test_unit
+      create    test/components/example_component_test.rb
+      invoke  erb
+      create    app/components/example_component.html.erb
+```
+
+## Options
+
+The component generator accepts several options to customize its behavior.
+
+### Override the default template engine
+
+ViewComponent includes template generators for the `erb`, `haml`, and `slim` template engines and will default to the template engine specified in `config.generators.template_engine`.
```

Comment:
🤔 should we add a similar guide for all `config` options?

Reply:
I was about to suggest that, yes!

Reply:
Awesome! I think it might make sense as part of the API docs. I'll look into it ❤️

Reply:
Can you create an issue for this? I may be able to tackle it.